napari/layers/_shapes_layer/model.py

```diff
@@ -259,10 +283,41 @@ def data(self):
 
     @data.setter
     def data(self, data):
+        if self._data == data:
+            return
         self._data = data
         self.events.data()
         self.refresh()
 
+    @property
+    def broadcast(self):
+        """Bool: if shapes are broadcast across all dimensions.
+        """
+        return self._broadcast
+
+    @broadcast.setter
+    def broadcast(self, broadcast):
+        if self._broadcast == broadcast:
+            return
+        self._broadcast = broadcast
+        self.events.broadcast()
+        self.refresh()
+
+    @property
+    def nshapes(self):
+        """int: total number of shapes.
+        """
+        nshapes = 0
+        for key, data in self.data.items():
+            nshapes = nshapes + len(data.shapes)
```

Comment:
I'd refactor this whole function as `nshapes = sum(len(data.shapes) for data in self.data.values())`

I actually don't understand why this isn't just `len(data)`, or rather,

```python
nshapes = sum(map(len, self.data.values()))
```

Unless I'm misreading the code, each element is a ShapeList, which should have its own `__len__` method and return the right number, right?

Reply:
Ah so a `ShapeList` isn't actually a `List`, if this is confusing and misleading I can try fixing with help from @kne42. It contains a property `shapes` which is the `List`. We can fix this, but it should be a followup PR

Reply:
I have replace this with `nshapes = sum(len(data.shapes) for data in self.data.values())` for now